bittensor/commands/metagraph.py

```diff
@@ -76,27 +77,31 @@ def run(cli: "bittensor.cli"):
             subtensor: "bittensor.subtensor" = bittensor.subtensor(
                 config=cli.config, log_verbose=False
             )
-            MetagraphCommand._run(cli, subtensor)
+            asyncio.run(MetagraphCommand._run(cli, subtensor))
```

Comment:
Since we are doing async work on the entire Bittensor package, it is assumed that all the code is async, include `%CommsndName%.run` functions.
The entry point to Bittensor SDK is the `btcli` utility. This is where we will create an event loop to execute/run/use async code.
@thewhaleking

Reply:
I actually disagree. I think that with the current architecture, each command is basically its own entrypoint. While I don't necessarily think that's the best architecture design, fixing it currently will delay our async efforts. Because we'd ideally not create a new subtensor object in each `run` method.

Reply:
IMHO
My approach is this:

1. In btcli we create an event loop
2. In btcli we create a Subtensor instance
3. We pass the Subtensor instance to the Commands.run() as a argument.
4. The same way the commander will do it.

In this case, we get:

1. a completely asynchronous SDK
2. the correct architecture, which will not have many sync entry points. 

This is best practices of the async impl.

Reply:
While I agree wholeheartedly that is the correct approach overall, I think getting the async stuff sorted first, and then going back and fixing the architecture is the best method, because of the 26 June deadline for this.

Reply:
I'm OK with this approach. 
Migrate the bittensor -> migrate the commands -> apply the abstract class to the commands (bring those to one controlled structure) -> change the use bittensor (sync -> async) in btcli.
Sounds like a great plan!